context/src/main/java/io/grpc/Context.java

```diff
@@ -99,6 +99,12 @@
   private static final PersistentHashArrayMappedTrie<Key<?>, Object> EMPTY_ENTRIES =
       new PersistentHashArrayMappedTrie<Key<?>, Object>();
 
+  // Long chains of contexts are suspicious and usually indicate a misuse of Context.
+  // The threshold and interval are arbitrarily chosen.
+  // VisibleForTesting
+  static final int CONTEXT_DEPTH_WARN_THRESH = 200;
+  private static final int CONTEXT_WARN_INTERVAL = 20;
```

Comment:
Let's not do the interval warning at this point. Since we've not been warning before, let's start slow.